src/core.js

```diff
@@ -2394,22 +2393,35 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
    */
   this.setSourceDataAtCell = function(row, column, value, source) {
     const input = setDataInputToArray(row, column, value);
-    const changes = [];
-
-    input.forEach((change, i) => {
-      const [changeRow, changeProp, changeValue] = change;
-      changes.push([
-        input[i][0],
-        input[i][1],
-        dataSource.getAtCell(input[i][0], input[i][1]),
-        input[i][2],
-      ]);
+    const isThereAnySetSourceListener = this.hasHook('afterSetSourceDataAtCell');
+    const changesForHook = [];
+
+    if (isThereAnySetSourceListener) {
+      arrayEach(input, ([changeRow, changeProp, changeValue]) => {
```

Comment:
Do you use helpers :)? Maybe native `Array.each` will be faster?

Reply:
`arrayEach` is more stable :) There is no policy about the usage of the native vs helpers. I'll leave as it is.